app/uk/gov/hmrc/ngrraldfrontend/controllers/ProvideDetailsOfSecondRentPeriodController.scala

```diff
@@ -0,0 +1,105 @@
+import uk.gov.hmrc.ngrraldfrontend.repo.SessionRepository
+import uk.gov.hmrc.ngrraldfrontend.utils.DateKeyFinder
+import uk.gov.hmrc.ngrraldfrontend.views.html.ProvideDetailsOfSecondRentPeriodView
+
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import play.api.libs.json.*
+
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.util.Locale
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class ProvideDetailsOfSecondRentPeriodController @Inject()(view: ProvideDetailsOfSecondRentPeriodView,
+                                                           authenticate: AuthRetrievals,
+                                                           mcc: MessagesControllerComponents,
+                                                           getData: DataRetrievalAction,
+                                                           sessionRepository: SessionRepository,
+                                                           navigator: Navigator,
+                                                         )(implicit appConfig: AppConfig, ec: ExecutionContext)
+  extends FrontendController(mcc) with I18nSupport with DateKeyFinder:
+
+
+
+  def show(mode: Mode): Action[AnyContent] =
+    (authenticate andThen getData).async { implicit request =>
+
+      def formatDate = {
+        val firstRentPeriodEnd = request.userAnswers.getOrElse(UserAnswers(request.credId))
+          .get(ProvideDetailsOfFirstRentPeriodPage).map(_.endDate).getOrElse("").toString
+        val date = LocalDate.parse(firstRentPeriodEnd)
+        val outputFormatter = DateTimeFormatter.ofPattern("d MMMM yyyy", Locale.UK)
+        date.format(outputFormatter)
+      }
+
+      formatDate match {
+        case "" => Future.successful(Redirect(routes.ProvideDetailsOfFirstRentPeriodController.show(NormalMode)))
+        case _ =>
+          val preparedForm = request.userAnswers.getOrElse(UserAnswers(request.credId)).get(ProvideDetailsOfSecondRentPeriodPage).fold(form)(form.fill)
+          Future.successful(Ok(view(
+            request.property.addressFull,
+            preparedForm,
+            formatDate,
+            endDateInput,
+            mode = mode
+            )))
+      }
+    }
+
+  def submit(mode: Mode): Action[AnyContent] =
+    (authenticate andThen getData).async { implicit request =>
+      form
+        .bindFromRequest()
+        .fold(
+          formWithErrors =>
+            def formatDate = {
+              val firstRentPeriodEnd = request.userAnswers.getOrElse(UserAnswers(request.credId))
+                .get(ProvideDetailsOfFirstRentPeriodPage).map(_.endDate).getOrElse(ProvideDetailsOfSecondRentPeriodPage).toString
+              val date = LocalDate.parse(firstRentPeriodEnd)
+              val outputFormatter = DateTimeFormatter.ofPattern("d MMMM yyyy", Locale.UK)
+              date.format(outputFormatter)
```

Comment:
```suggestion
              NGRDate.formatDate(firstRentPeriodEnd)
```
Line 87 to 89 can be replaced by this